### ISSUE TITLE
About extrinsics and intrinsics 

### ISSUE THREAD

**Issue by chendaozai** [saic-vul/imvoxelnet#58]
Hi! I have trouble projecting image features to voxels. I hope you can give me help, please.
I load intrinsics and extrinsics  from **intrinsic_color.txt** and **image_num.txt**. In addition, I read axis align matrix from scene_num.txt.
Then I copmute projection matrix according to the following equation：
**projection_matrix = intrinsic @ ( axis_align_matrix @ extrinsics )^(-1)**
At last, I run the function  backproject(feats, points, projection) to get volume.
However, the projection does not seem to be accurate.  
So, do I miss part of the processing of extrinsics and intrinsics? Do I need to axis align points?

**Comment by filaPro** [comment]
Following these 2 lines [1](https://github.com/saic-vul/imvoxelnet/blob/master/mmdet3d/datasets/scannet_monocular_dataset.py#L24), [2](https://github.com/saic-vul/imvoxelnet/blob/master/mmdet3d/models/detectors/imvoxelnet.py#L128) your matrix should be fine.

**Comment by chendaozai** [comment]
> Following these 2 lines [1](https://github.com/saic-vul/imvoxelnet/blob/master/mmdet3d/datasets/scannet_monocular_dataset.py#L24), [2](https://github.com/saic-vul/imvoxelnet/blob/master/mmdet3d/models/detectors/imvoxelnet.py#L128) your matrix should be fine.

Thank you for your reply, but the problem remains. My code is as follows：
`
# load intrinsic
intrinsic_path = './data/image_data/scannet_images/intrinsic/intrinsic_color.txt'
intrinsic = torch.tensor(np.loadtxt(intrinsic_path)[:3, :3])

# load axisAligment
axisAligment_path = './data/image_data/scannet_images/scene0000_00.txt'
lines = open(axisAligment_path).readlines() 
axis_align_matrix = np.eye(4)
intrinsic_data = []
for line in lines:
    if 'axisAlignment' in line:
        axis_align_matrix = [
            float(x)
            for x in line.rstrip().strip('axisAlignment = ').split(' ')
        ]
    elif '_color = ' in line:
        intrinsic_data.append(float(line.split('_color = ')[-1].rstrip()))
tmp = intrinsic_data[0]
intrinsic = torch.DoubleTensor([
        [intrinsic_data[0], 0 , intrinsic_data[2]],
        [0, intrinsic_data[1], intrinsic_data[3]],
        [0, 0, 1]])
axis_align_matrix = np.array(axis_align_matrix).reshape((4, 4))

# load pth file
p_path = "./data/image_data/scannetv2/train/scene0000_00_inst_nostuff.pth"
ori_coords, ori_colors, _, _ = torch.load(p_path)
colors = copy.deepcopy(ori_colors)

# random sapmle 80 masks and pose matrices of image（stride 70）
# sample one just test demo, and regard rgb as feats
pose_path = './data/image_data/scannet_images/pose/55.txt'
img_path = './data/image_data/mask_images/55.png'
pose = np.loadtxt(pose_path)

# Axis alignment
pose = torch.tensor(np.linalg.inv(axis_align_matrix @ pose))
mask_img = cv2.imread(img_path)
## simulation model process
mask_img = cv2.resize(mask_img, None, fx=0.25, fy=0.25)
cv2.imwrite('mask.jpg', mask_img)
feats = torch.tensor(mask_img.astype(np.float32)).permute(-1, 0, 1).unsqueeze(0) 
## Binarization mask
feats = torch.where(feats>122, 255, 0).to(torch.float32)

# compute projection matrix
ratio = 4
intrinsic[:2] /= ratio

projection = torch.stack([intrinsic @ pose[:3]])

# get points and voxel mask
origin = np.array([.0, .0, .0]).astype(np.float32)
n_voxels=np.array((120, 120, 50))
voxel_size = np.array([.08, .08, .08])
points = get_points(torch.tensor(n_voxels), torch.tensor(voxel_size), torch.tensor(origin))
points, volume, valid = backproject(feats, points, projection) # batch * (feats/coords/flag) * hwd
`
The real image and mask image are as follows:
![34](https://user-images.githubusercontent.com/55949813/192128355-f052c1a4-5c92-46ac-b142-60126f0d3d52.jpg)
![34](https://user-images.githubusercontent.com/55949813/192128351-0f90b2b1-0068-4a09-85fe-26401e42cf36.png)
The visualisation of volume is as follows:
![image](https://user-images.githubusercontent.com/55949813/192128441-9fde7181-7b18-4c0c-9795-93f780a8c606.png)
The red box is the content of the image and the blue section is the area projected by the camera. Unfortunately, the two do not correspond to each other.
